viewer/viewer-awt/src/main/java/org/icepdf/ri/common/SwingController.java

```diff
@@ -280,6 +284,8 @@
     protected static ResourceBundle messageBundle = null;
 
     protected ViewerPropertiesManager propertiesManager;
+    private Map<Reference, StateManager.Change> savedChanges = new HashMap<>();
```

Comment:
I spend some time revoing this tonight.  It's all very slick, thanks for the work.  My only comment for discussion is the savedChanges State.  I'm not  a huge fan of putting the state in the controller.  Would you consider a small api tweak to keep this state in the stateManager with some methods to keep it all well defined?